### PR TITLE
OLMIS-7659: Implement product info page

### PR DESCRIPTION
**Pull request by mdulko**
image

### CODE REVIEW

File: src/stock-on-hand-mobile/stock-on-hand-app.jsx [hunk 1]
```diff
@@ -21,6 +21,8 @@ import { setSupervisedProgramsStockOnHand } from './reducers/programs';
 import { setUserHomeFacilityStockOnHand, setSupervisedFacilitiesStockOnHand } from './reducers/facilities';
 import ProgramSelect from './pages/program-select';
 import StockOnHand from './pages/stock-on-hand';
+import ProductInfo from "./pages/product-details";
+import ProductDetails from "./pages/product-details";
```
Comment (dariapionk) [inline src/stock-on-hand-mobile/stock-on-hand-app.jsx:25]: Please change from double quote to single quote

File: src/stock-on-hand-mobile/components/stock-on-hand-table-component.jsx [hunk 2]
```diff
@@ -5,30 +5,35 @@
...
-}) => {
+                              columns,
+                              data,
+                              facility,
+                              program,
+                              hiddenColumns,
+                              isProductExpanded,
+                              expandedProducts,
+                              ...props
+                          }) => {
```
Comment (dariapionk) [inline src/stock-on-hand-mobile/components/stock-on-hand-table-component.jsx:31]: please fix the indentation

File: src/stock-on-hand-mobile/components/stock-on-hand-table-component.jsx [hunk 3]
```diff
@@ -37,7 +42,20 @@ const StockOnHandTable = ({
...
+
+    const history = useHistory();
+    const dispatch = useDispatch();
+
+    const goToProductInfo = (product) => {
+        const facilityId = facility.id
+        const programId = program.id
+        const productId = product.orderable.id
+
+        dispatch(setProductData(product))
```
Comment (sniedzielski) [inline src/stock-on-hand-mobile/components/stock-on-hand-table-component.jsx:54]: missing semicolons:
46-47,
50-52,
54
lines

File: src/stock-on-hand-mobile/pages/product-details.jsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            return (
+                date.getDate()+"/"+(date.getMonth()+1)+"/"+date.getFullYear()
+            )
+        }
+        else {
+            const correctDate = new Date(date);
+
+            return (
+                correctDate.getDate()+"/"+(correctDate.getMonth()+1)+"/"+correctDate.getFullYear()
+            )
```
Comment (sniedzielski) [inline src/stock-on-hand-mobile/pages/product-details.jsx:39]: missing semicolon

File: src/stock-on-hand-mobile/pages/product-details.jsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+    const product = useSelector(state => state['productStockOnHand']['productData']);
+    const facility = useSelector(state => state['facilitiesStockOnHand']['facilityStockOnHand']);
+    const program = useSelector(state => state['programsStockOnHand']['programStockOnHand']);
+
+    const dateFormat = (date) => {
+        if (typeof date !== 'string'){
+            return (
+                date.getDate()+"/"+(date.getMonth()+1)+"/"+date.getFullYear()
+            )
```
Comment (sniedzielski) [inline src/stock-on-hand-mobile/pages/product-details.jsx:32]: missing semicolon

File: src/stock-on-hand-mobile/pages/product-details.jsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+const ProductDetails = () => {
+
+    const history = useHistory();
+
+    const product = useSelector(state => state['productStockOnHand']['productData']);
+    const facility = useSelector(state => state['facilitiesStockOnHand']['facilityStockOnHand']);
+    const program = useSelector(state => state['programsStockOnHand']['programStockOnHand']);
+
+    const dateFormat = (date) => {
+        if (typeof date !== 'string'){
```
Comment (sniedzielski) [inline src/stock-on-hand-mobile/pages/product-details.jsx:29]: `        if (typeof date !== 'string') {`
please add whitespace between curly bracket and 'if' parentheses

File: src/stock-on-hand-mobile/pages/product-details.jsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+            )
+        }
+
+    }
+
+    const productData = [
+        { name: 'Stock on Hand', value: product?.stockOnHand },
+        { name: 'Product Code', value: product?.orderable?.productCode },
+        { name: 'Lot Code', value: product?.lot?.lotCode ?? 'No lot defined' },
+        { name: 'Expiry Date', value: product?.lot ? dateFormat(product?.lot?.expirationDate) : 'No lot defined'  },
```
Comment (sniedzielski) [inline src/stock-on-hand-mobile/pages/product-details.jsx:48]: `        { name: 'Expiry Date', value: product?.lot ? dateFormat(product?.lot?.expirationDate) : 'No lot defined' },`
the ending has one redundant whitespace. Please remove to have like above.

**Review by sniedzielski — COMMENTED** [review]
I left some comments, please refer to them @mdulko

File: src/stock-on-hand-mobile/components/stock-on-hand-table-component.jsx (hunk 2 above)
Comment (mdulko) [reply src/stock-on-hand-mobile/components/stock-on-hand-table-component.jsx]: updated

File: src/stock-on-hand-mobile/components/stock-on-hand-table-component.jsx (hunk 3 above)
Comment (mdulko) [reply src/stock-on-hand-mobile/components/stock-on-hand-table-component.jsx]: updated

**Review by sniedzielski — APPROVED** [review]
LGTM @mdulko . @dariapionk Do you accept those changes applied by Michal?

**Comment by dariapionk** [comment]
LGTM too @mdulko

**Comment by sniedzielski** [comment]
Ok. I will merge this PR.

**Pull request merged**